src/spade/transformer/ContextualTemporalTraversal.java

```diff
@@ -0,0 +1,272 @@
+                HashMap<String, AbstractEdge> vertexParentEdges = edgeMap.get(vertex.bigHashCode()).get("parentEdges");
+
+                for (HashMap.Entry<String, AbstractEdge> entry : vertexParentEdges.entrySet()) {
+                    AbstractEdge edge = entry.getValue();
+                    AbstractEdge newEdge = createNewWithoutAnnotations(edge);
+                    try {
+                        Double time = Double.parseDouble(getAnnotationSafe(newEdge, annotationName));
+                        if (minTime == null || time < minTime) {
+                            minTime = time;
+                        }
+                    } catch (Exception e) {
+                        logger.log(Level.SEVERE, "Failed to parse where " + annotationName + "='" + getAnnotationSafe(newEdge, annotationName) + "'");
+                    }
+                }
+        
+                if (minTime == null) {
+                    // Log that no valid timestamp was found
+                    logger.log(Level.INFO, "No valid timestamp found for vertex: " + vertex);
+                }
+                return minTime;
+            }
+
+        /*
+         * Get children for a vertex that are from edges with increasing time, edges with lesser time will not be considered
+         * Note: Passing adjacentGraph to collect all children from multiple vertices from one level into one graph
+         */
+        public Graph getAllChildrenForAVertex(AbstractVertex vertex, Graph adjacentGraph, Graph finalGraph, Graph graph, Integer levelCount, HashMap<String, HashMap<String, HashMap<String, AbstractEdge>>> edgeMap, HashMap<String, HashMap<String, HashMap<String, AbstractEdge>>> finalEdgeMap) {
+                Double minTime = null;
+                Graph childGraph = finalGraph.vertexSet().isEmpty() || adjacentGraph == null ? new Graph() : adjacentGraph;
+                minTime = finalGraph.vertexSet().isEmpty() ? getMintime(vertex, edgeMap) : getMintime(vertex, finalEdgeMap);
+                if (levelCount == 1) {
+                    if (minTime != null && (graphMinTime == null || minTime < graphMinTime)) {
+                        graphMinTime = minTime;
+                    }
+                }
+                logger.log(Level.INFO, "Got mintime: " + minTime);
+                HashMap<String, AbstractEdge> vertexChildEdges = edgeMap.get(vertex.bigHashCode()).get("childEdges");
+                HashMap<String, AbstractEdge> vertexParentEdges = edgeMap.get(vertex.bigHashCode()).get("parentEdges");
+        
+                for (HashMap.Entry<String, AbstractEdge> entry : vertexChildEdges.entrySet()) {
+                    AbstractEdge edge = entry.getValue();
+                    AbstractEdge newEdge = createNewWithoutAnnotations(edge);
+                    try {
+                        Double time = Double.parseDouble(getAnnotationSafe(newEdge, annotationName));
+                        if (minTime == null || time > minTime) {
```

Comment:
Just for clarity, can you please add a comment here saying something like 'if minTime is not found then all child edges are considered adjacent'? Thank you.

Reply:
Done